src/scripts/main.js

```diff
@@ -1,3 +1,256 @@
+
+  notification.className = `notification ${type}`;
+  notification.setAttribute('data-qa', 'notification');
+
+  titleSpan.className = 'title';
+  titleSpan.textContent = type.toUpperCase();
+
+  notification.appendChild(titleSpan);
+  notification.innerHTML += ` <span>${message}</span>`;
+  document.body.appendChild(notification);
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, 2000);
+}
+
+function validateFormData(formData) {
+  const nameRegex = /^[A-Za-z\s]+$/;
+
+  if (formData.name.length < 4 || !nameRegex.test(formData.name)) {
+    pushNotification(
+      'Name must be at least 4 characters long and contain only letters.',
+      'error',
+    );
+
+    return false;
+  }
+
+  if (formData.age < 18 || formData.age > 90) {
+    pushNotification('Age must be between 18 and 90.', 'error');
+
+    return false;
+  }
+
+  if (!formData.position.trim()) {
+    pushNotification('Position is required.', 'error');
+
+    return false;
+  }
+
+  if (!formData.office.trim()) {
+    pushNotification('Office is required.', 'error');
+
+    return false;
+  }
+
+  const salary = parseFloat(formData.salary);
+
+  if (isNaN(salary) || salary <= 0) {
+    pushNotification('Salary must be a valid positive number.', 'error');
+
+    return false;
+  }
+
+  return true;
+}
+
+const form = document.querySelector('.new-employee-form');
+
+document.getElementById('save-button').addEventListener('click', () => {
+  const formData = [...form.elements].reduce((acc, formInput) => {
+    if (formInput.name) {
+      acc[formInput.name] = formInput.value;
+    }
+
+    return acc;
+  }, {});
+
+  if (validateFormData(formData)) {
+    const newRow = document.createElement('tr');
+
+    newRow.innerHTML = `
+      <td>${formData.name}</td>
+      <td>${formData.position}</td>
+      <td>${formData.office}</td>
+      <td>${formData.age}</td>
+      <td>$${parseInt(formData.salary).toLocaleString('en-US')}</td>`;
+
+    table.querySelector('tbody').appendChild(newRow);
+    pushNotification('Employee added successfully!', 'success');
+    form.reset();
+  }
+});
+
+// 5. Implement editing of table cells by double-clicking on it. (optional)
+const input = document.createElement('input');
+
+input.className = 'cell-input';
+
+input.addEventListener('blur', () => {
+  const cell = input.parentElement;
+
+  cell.innerText = input.value || input.getAttribute('data-initial');
+});
+
+input.addEventListener('keypress', (e) => {
+  if (e.key === 'Enter') {
+    const cell = input.parentElement;
+
+    cell.innerText = input.value || input.getAttribute('data-initial');
+  }
+});
+
+document.querySelectorAll('tbody td').forEach((cell) => {
+  cell.addEventListener('dblclick', () => {
+    const initialValue = cell.innerText;
+
+    input.setAttribute('data-initial', initialValue);
+    input.value = initialValue;
+
+    cell.innerText = '';
+    cell.appendChild(input);
+    input.focus();
+  });
+});
```

Comment:
Bug: Similar to the issue with the click event listener for the rows, the double-click event listener for the cells is added only once when the script is loaded. If new cells are added to the table afterwards, they won't have the double-click event listener. Consider adding the event listener each time a new cell is added.